vizier/ui/src/info/vizierdb/ui/components/ChartEditor.scala

```diff
@@ -0,0 +1,745 @@
+package info.vizierdb.ui.components.editors
+
+import info.vizierdb.ui.components._
+import info.vizierdb.ui.rxExtras.implicits._
+import info.vizierdb.ui.widgets.FontAwesome
+import info.vizierdb.ui.widgets.Spinner
+import info.vizierdb.serialized.FilesystemObject
+import info.vizierdb.types.MIME
+import info.vizierdb.ui.Vizier
+import info.vizierdb.types.DatasetFormat
+import play.api.libs.json._
+import org.scalajs.dom
+import scalatags.JsDom.all._
+import rx._
+import info.vizierdb.serialized
+import scala.concurrent.ExecutionContext.Implicits.global
+import info.vizierdb.util.Logging
+import info.vizierdb.serializers._
+import info.vizierdb.api.websocket
+import info.vizierdb.types.ArtifactType
+import info.vizierdb.serialized.{
+  CommandArgument,
+  CommandArgumentList,
+  CommandDescription,
+  ParameterDescriptionTree,
+  DatasetSummary,
+  DatasetDescription,
+  ArtifactDescription,
+  DatasetColumn,
+  PackageCommand,
+  PropertyList
+}
+import info.vizierdb.ui.rxExtras.{OnMount, RxBuffer, RxBufferView}
+import info.vizierdb.types._
+import info.vizierdb.nativeTypes.JsValue
+import scala.util.{Success, Failure}
+import info.vizierdb.ui.network.BranchSubscription
+import info.vizierdb.ui.network.BranchWatcherAPIProxy
+import info.vizierdb.ui.widgets.FontAwesome
+import java.awt.Font
+import scala.concurrent.Future
+import info.vizierdb.serialized.CommandArgument
+import info.vizierdb.ui.components.dataset.Dataset
+import info.vizierdb.serialized.EnumerableValueDescription
+import info.vizierdb.artifacts.VegaRegressionMethod
+
+
+class ChartEditor(
+    override val delegate: ModuleEditorDelegate,
+    override val packageId: String,
+    override val command: serialized.PackageCommand,
+    val chartType: String
+)(implicit owner: Ctx.Owner)
+    extends DefaultModuleEditor(packageId, command, delegate)
+    with Logging {
+    // Keeps State of the arguments after editing
+    println(chartType)
```

Comment:
Use logger instead of println for logging messages that get checked in.  These are easier to turn on/off.